reco_analysis/reco_analysis/chatbot/chatbot.py

```diff
@@ -228,6 +255,9 @@ def generate_response(self) -> str:
         # Save the AI's response to the memory
         self.send(response)
 
+        # Detect and handle the end of the conversation if the role is Patient
+        self.detect_and_handle_end() if self.role == 'Patient' else None
```

Comment:
I think it might be wise to only let Doctor agent make the call on whether or not to terminate? no point invoking the same prompt twice, right?

Reply:
Yes removed.